### PR TITLE
Modified EntityLIving.depawnEntity()

### PR DESCRIPTION
**Pull request by MrChoke72**
This is the 2nd solution to fix issues #5148 and #4485.  It is better than the 1st in that it ensures that the forge LivingEvent.AllowDespawn() event handler is not called more often than every 32 ticks.

As explained in the previous solution, the current implementation in Forge does not allow full depawn override capability for the modder.  Many parts of the default behavior like distance checks will still execute and despawn entities regardless of what the modder wants.  That is because 31 out of every 32 ticks, the default behavior is called.

This solution will only call the AllowDespawn event handler once every 32 ticks.  Below is the behavior to the handler's result:
DENY:  The event handler AND the default behavior are both not executed for 32 ticks.
ALLOW:  The entity is despawned.
DEFAULT:  The default behavior is called for 32 ticks without calling the event handler

The last solution had a performance concern with the even the handler being called every tick.  As you can see, this design addresses that concern.

### CODE REVIEW

**Comment by liach** [comment]
be careful, no imports in patches

**Comment by MrChoke72** [comment]
> be careful, no imports in patches


> be careful, no imports in patches

Oh yeah, that was from me debugging..  Oops.  I will remove it.  Thanks.

File: patches/minecraft/net/minecraft/entity/EntityLiving.java.patch [hunk 1]
```diff
@@ -1,14 +1,23 @@
 --- ../src-base/minecraft/net/minecraft/entity/EntityLiving.java
 +++ ../src-work/minecraft/net/minecraft/entity/EntityLiving.java
-@@ -170,6 +170,7 @@
+@@ -84,6 +84,8 @@
+     private boolean field_110169_bv;
+     private Entity field_110168_bw;
+     private NBTTagCompound field_110170_bx;
++    protected int despawnHndlrWaitTime;
```
Comment (MajorTuvok) [inline patches/minecraft/net/minecraft/entity/EntityLiving.java.patch]: Typo... (Handler)

File: patches/minecraft/net/minecraft/entity/EntityLiving.java.patch [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
++    protected boolean forceNodespawn;
```
Comment (diesieben07) [inline patches/minecraft/net/minecraft/entity/EntityLiving.java.patch]: Should be spelled `forceNoDespawn`, otherwise it reads "node" + "spawn".

**Comment by MrChoke72** [comment]
Ok.  The two variable naming suggestions have been applied.  Thanks.

**Review by Alpvax — APPROVED** [review]

**Comment by MrChoke72** [comment]
Guys, I just pushed a change to reset forceNoDespawn back to false if the event handler comes back DEFAULT.  I swear I had that in there before but oh well.  It's there now.

**Review by tterrag1098 — CHANGES_REQUESTED** [review]
Lots of extra unnecessary changes in this commit

**Comment by MrChoke72** [comment]
I am replying to stop this PR from being "stale".  I applied all pending review changes.  I think it's a good change for 1.12.

**Pull request closed without merging**